app/models/flat_reasons_for_rejection_extract.rb

```diff
@@ -0,0 +1,183 @@
+class FlatReasonsForRejectionExtract
```

Comment:
I wonder if this class could replace / merge with the existing `ReasonsForRejection` model?

Reply:
Echoing what @despo has said: 
Given this is a unpersisted/transient set of attributes on a model would we be better to turn this into an export-centric [Presenter](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/presenters/rejected_application_choice_presenter.rb) which delegates to the underlying [`ReasonsForRejection`](https://github.com/DFE-Digital/apply-for-teacher-training/blob/ebb0d63a0a32507206cf3a29ee30c7bdc66492e7/app/models/reasons_for_rejection.rb) model attributes?
It seems the main concern here is to extract/translate/present what we store in `ApplicationChoice#structured_reasons_for_rejection` but there's a lot of repetition of attributes which are already defined in `ReasonsForRejection` and presentation seems to be the goal here.
I don't fully understand the need for another model-like class as opposed to something which presents the data in a desireable format.

Reply:
@steventux I think the intention here is to make something that's a bit more generic than what's needed just for the export in this PR because this kind of requirement has come up a few times. But a presenter makes sense to me too.

Reply:
I'll have a proper look at this over the weekend, but yea a presenter does seem to make sense.

Reply:
Yea my main thoughts is that the naming of this file is a bit misleading. as Steve said the purpose of this is that we use the nested data in the structure in multiple exports and throughout the dashboard. 

We need an easy way to replicate each individual checkbox that the provider selects. I think having an easy way to return nested values as a flat object would be pretty useful so we don't have to use stuff like `application_choice.structured_rejection_reasons.dig('qualifications_which_qualifications'], 'no_maths_gcse')` in multiple places in the export (for each attr).

I agree though that it is essentially taking the data we've got and presenting it in a different way though and could/should live in a presenter.

Reply:
It's worth noting you can pass the data from the application choice into the `ReasonsForRejection` pseudo-model like this https://github.com/DFE-Digital/apply-for-teacher-training/blob/3dcd4625615b959488b7c5043453a32b1936d1ac/app/components/provider_interface/application_rejection_feedback_component.html.erb#L13 and then interrogate/present the attributes in a more OO way